### PR TITLE
Ask users for PI at registration; abstract user registration code into a single script

### PR DESCRIPTION
**Pull request by yantosca**
This PR does the following:

1. Creates a new script (`run/shared/newUserRegistration.sh`) that contains the common bash functions for user registration.  This allows us to make modifications to the user registration process in a single location that can be called from the GCC or GCHP `createRunDir.sh` script.
2. Updates the user registration process to ask "What is the name of your research institution?" (as per Randall's suggestion)
3. Updates the user registration process to ask "What is the name of your PI?" (as per Randall's suggestion)
4. Defines a user prompt `>>> ` that is used for not only the user registration but also GEOS-Chem rundir creation questions.  For consistency's sake.

NOTE: @laestrada will need to make PR to the gc-cloud-infrastructure to add some processing on the AWS side to be able to add the new field into the database.  But for now the script is succesffuly able to push information to the AWS cloud.

The new user registration screen now looks like this:
```console
===========================================================
GEOS-CHEM RUN DIRECTORY CREATION
===========================================================

Initiating User Registration:
You will only need to fill this information out once.
Please respond to all questions.

-----------------------------------------------------------
What is your name?
-----------------------------------------------------------
>>> Homer J Simpson

-----------------------------------------------------------
What is your email address?
-----------------------------------------------------------
>>> homerj@snpp.com

-----------------------------------------------------------
What is the name of your research institution?
-----------------------------------------------------------
>>> Springfield Nuclear Power Plant

-----------------------------------------------------------
What is the name of your principal invesigator?
(Enter 'self' if you are the principal investigator.)
-----------------------------------------------------------
>>> Mr. Burns

-----------------------------------------------------------
Please provide the web site for your institution
(group website, company website, etc.)?
-----------------------------------------------------------
>>> snpp.com

-----------------------------------------------------------
Please provide your github username (if any) so that we
can recognize you in submitted issues and pull requests.
-----------------------------------------------------------
>>> @homerj

-----------------------------------------------------------
Where do you plan to run GEOS-Chem?
(e.g. local compute cluster, AWS, other supercomputer)?
-----------------------------------------------------------
>>> local compute cluster

-----------------------------------------------------------
Please briefly describe how you plan on using GEOS-Chem
so that we can add you to 'GEOS-Chem People and Projects'
(https://geoschem.github.io/geos-chem-people-projects-map/)
-----------------------------------------------------------
>>> testing and development
Successful Registration
```

Also, the email is the primary key.  If a user re-registers with the same email, the prior registration information on file will be updated.

### CODE REVIEW

**Comment by yantosca** [comment]
All GEOS-Chem Classic integration tests passed:
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #76f9571 GEOS-Chem submod update: Merge PRs #1663, #1669 into version 14.1.1
GEOS-Chem #a9c80b53b Fix issues with new user registration; update registration questions
HEMCO     #cf7d4ed GEOS-only updates by Christoph Keller

Using 24 OpenMP threads
Number of execution tests: 26

Submitted as SLURM job: 44152836
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gc_05x0625_NA_47L_merra2_CH4........................Execute Simulation....PASS
gc_05x0625_NA_47L_merra2_fullchem...................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem..........................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem_TOMAS15..................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem_TOMAS40..................Execute Simulation....PASS
gc_4x5_merra2_aerosol...............................Execute Simulation....PASS
gc_4x5_merra2_carbon................................Execute Simulation....PASS
gc_4x5_merra2_CH4...................................Execute Simulation....PASS
gc_4x5_merra2_CO2...................................Execute Simulation....PASS
gc_4x5_merra2_fullchem..............................Execute Simulation....PASS
gc_4x5_merra2_fullchem_aciduptake...................Execute Simulation....PASS
gc_4x5_merra2_fullchem_APM..........................Execute Simulation....PASS
gc_4x5_merra2_fullchem_benchmark....................Execute Simulation....PASS
gc_4x5_merra2_fullchem_complexSOA...................Execute Simulation....PASS
gc_4x5_merra2_fullchem_complexSOA_SVPOA.............Execute Simulation....PASS
gc_4x5_merra2_fullchem_LuoWd........................Execute Simulation....PASS
gc_4x5_merra2_fullchem_marinePOA....................Execute Simulation....PASS
gc_4x5_merra2_fullchem_RRTMG........................Execute Simulation....PASS
gc_4x5_merra2_Hg....................................Execute Simulation....PASS
gc_4x5_merra2_metals................................Execute Simulation....PASS
gc_4x5_merra2_POPs_BaP..............................Execute Simulation....PASS
gc_4x5_merra2_tagCH4................................Execute Simulation....PASS
gc_4x5_merra2_tagCO.................................Execute Simulation....PASS
gc_4x5_merra2_tagO3.................................Execute Simulation....PASS
gc_4x5_merra2_TransportTracers......................Execute Simulation....PASS
gc_4x5_merra2_TransportTracers_LuoWd................Execute Simulation....PASS
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 26
Execution tests failed: 0
Execution tests not yet completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
All GCHP integration tests except tagO3 (which is a known issue) passed.
```console
==============================================================================
GCHP: Execution Test Results

GCClassic #8d3db1a GEOS-Chem submod update: Merge PRs #1663, #1669 into version 14.1.1
GEOS-Chem #a9c80b53b Fix issues with new user registration; update registration questions
HEMCO     #cf7d4ed GEOS-only updates by Christoph Keller

Number of execution tests: 5

Submitted as SLURM job: 44165818
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gchp_merra2_fullchem................................Execute Simulation....PASS
gchp_merra2_fullchem_benchmark......................Execute Simulation....PASS
gchp_merra2_fullchem_RRTMG..........................Execute Simulation....PASS
gchp_merra2_tagO3...................................Execute Simulation....FAIL
gchp_merra2_TransportTracers........................Execute Simulation....PASS
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 4
Execution tests failed: 1
Execution tests not yet completed: 0
```

**Review by lizziel — APPROVED** [review]

**Review by msulprizio — CHANGES_REQUESTED** [review]

**Comment by yantosca** [comment]
Also slipping into this branch a minor fix for GCHP integration test compilations in commit 9e0119343.  We remove "tagO3" from the `EXE_GCHP_BUILD_LIST` variable since tagO3 can use the default build.  This will reduce overall compilation time for GCHP integration tests.

File: run/GCHP/createRunDir.sh [hunk 1]
```diff
@@ -1,50 +1,47 @@
...
+#\\
+#\\
+# !CALLING SEQUENCE:
+#  ./createRunDir.sh [rundirname]
 #
-# Usage: ./createRunDir.sh [rundirname]
+# !REMARKS:
+#  If optional run directory name argument is not passed then the user
+#  will be prompted to enter a name interactively, or choose to use the
+#  default name gc_{grid_display}_{met}_{sim_name}_{sim_extra_option}.
```
[the remark this answers is not in the document]
Comment (yantosca) [reply run/GCHP/createRunDir.sh]: Good catch @msulprizio!  Now fixed in 2b4ac7983a33b06de5cab4bda14a07372d969b12

File: run/shared/newUserRegistration.sh [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+# User prompt for reading in data
+USER_PROMPT=">>> "
+
+####=========================================================================
+#### Send registration details to AWS database via curl
+####==========================================================================
+function postRegistration() {
+    #### Sends registration details to API ####
+    curl --location --request POST "https://gc-dashboard.org/registration" \
```
[the remark this answers is not in the document]
Comment (yantosca) [reply run/shared/newUserRegistration.sh:33]: Not yet @msulprizio.  I can also look into that.  @lucas said he would make a PR and tag me but if I can do it sooner then I will.

File: run/GCClassic/createRunDir.sh [hunk 3]
```diff
@@ -1,50 +1,48 @@
...
+#  See the subsequent Git history with the gitk browser!
+#------------------------------------------------------------------------------
+#BOC
+
+# Directory with GEOS-Chem Classic rundir scripts (i.e. this dir)
+# Current directory
 srcrundir=$(pwd -P)
 cd ${srcrundir}
+
+# GEOS-Chem "science codebase"directory
```
[the remark this answers is not in the document]
Comment (yantosca) [reply run/GCClassic/createRunDir.sh]: Now fixed in a96511761

**Review by msulprizio — APPROVED** [review]
This PR now looks good to merge!

**Comment by yantosca** [comment]
Thanks @msulprizio and @lizziel!  I will also try to make the changes on the gc-cloud-infrastructure side.

**Pull request merged**